static/src/components/site-header/site-header.ts

```diff
@@ -245,20 +343,40 @@ export class SiteHeaderDropdown {
         this.toggle.setAttribute("aria-expanded", `${this.expanded}`);
         this.list.setAttribute("aria-hidden", `${!this.expanded}`);
         this.list.style.height = this.expanded ? `${this.list.scrollHeight}px` : null;
+        const bodyElement = document.body;
         for (let i = 0; i < this.childLinks.length; i++) {
             this.childLinks[i].setAttribute("tabindex", this.expanded ? "0" : "-1");
         }
         if (this.expanded) {
+            bodyElement.style.overflow = 'hidden';
             this.events.open.forEach(handler => {
                 handler(this);
             });
         } else {
+            bodyElement.style.overflow = 'auto';
             this.events.close.forEach(handler => {
                 handler(this);
             });
         }
     }
 
+    protected handleCheckOverlayHeight () {
+        const checkOverlayHeight = () => {
+            const nestedUl = this.element.querySelector('.site-header__nav>ul>li ul') as HTMLElement;
+            if (window.innerWidth > 767) {
+                if (
+                    nestedUl.offsetHeight + this.list.offsetHeight >
```

Comment:
This feels like the wrong calculation to me.  What we are trying to figure out is if the overlay exceeds the page height.  This reads as though we are combining the `ul` height with another `ul` height.  What we want to know if the height of the overlay (I believe that is nestedUl.offsetHeight) and the distance from the [top of the window ](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect)that the overlay is.

Reply:
Updated while reviewing with you